pkgs/development/compilers/gcc/common/configure-flags.nix

```diff
@@ -229,6 +229,14 @@ let
       # On Illumos/Solaris GNU as is preferred
       "--with-gnu-as" "--without-gnu-ld"
     ]
+    ++ lib.optionals (targetPlatform.libc == "serenity") [
+      "--enable-initfini-array"
+      "--enable-threads=posix"
+      "--with-linker-hash-style=gnu"
+      "--disable-nls"
+      # provided by libc
+      "--disable-libssp"
```

Comment:
Maybe this could be merged into the NetBSD case above? Unsure whether that would be better.